airframe-sql/src/main/scala/wvlet/airframe/sql/analyzer/TypeResolver.scala

```diff
@@ -379,7 +385,27 @@ object TypeResolver extends LogSupport {
           case a: Attribute   => a
           case m: MultiColumn => m
           // retain alias for select column
-          case expr => if (isSelectItem) SingleColumn(expr, Some(i.value), None, expr.nodeLocation) else expr
+          case f: FunctionCall =>
+            // Extract source columns
+            val src = Set.newBuilder[SourceColumn]
```

Comment:
SourceColumn has TableCatalog as well, so using Set operation might be a bit heavy.